lib/parsers/jade.js

```diff
@@ -34,24 +34,42 @@ function parseJade(str, options) {
     return extractGettext(this[key].val);
   }
 
-  function isEmpty(obj) {
-    return obj.length;
+  var buf = [], lineN, lineAdjust = -1;
+
+  function append(text) {
+    /* jshint -W040 */
+    var ctx = this;
+    if (ctx && ctx.bump) {
+      lineN += 1;
+    }
+    if (text.length) {
+      buf[lineN] = [buf[lineN], text, ';'].join('');
```

Comment:
Better for me? :-)

    buf[lineN] = (buf[lineN] || '') + text + ';';

If you insist I can change it: I just think version with `join` seems cleaner.